phac-nml/irida-next
# GraphQL: Namespace Project linkage

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR updates our GraphQL api to be able to query Projects belonging to a Namespace or belonging to SubGroups when the Namespace is a Group. This also removes the Namespace interface and replaces it with a Namespace type. This change means that you cannot retrieve Group specific info when fetching the Namespace that a Project belongs too in a single query. Making this change means that we did not have to add a specific type for UserNamespaces.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as `admin@email.com`
1. Open the graphiql interface `http://localhost:3000/graphiql`
1. Query for a Namespace that you can access and request the projects within it.
   * ```graphql
     {
       namespace(fullPath: "bacillus/bacillus-anthracis") {
         id
         name
         projects {
           nodes {
             id
             name
           }
         }
       }
     }
     ```
1. See that you get a valid result
1. Query for a Namespace that you can access but only has projects through sub groups.
   * ```graphql
     {
       namespace(fullPath: "bacillus") {
         id
         name
         projects(includeSubGroups: true) {
           nodes {
             id
             name
           }
         }
       }
     }
     ```
1. See that you get a valid result

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
